docs/developer-docs/web-apps/verifiable-credentials/issuer.mdx

```diff
@@ -0,0 +1,184 @@
+import useBaseUrl from "@docusaurus/useBaseUrl";
+
+# Become an issuer
+
+<MarkdownChipRow labels={["Intermediate", "Tutorial"]} />
+
+## Overview
+
+An **issuer** is a service or app that can issue a verifiable credential (VC) to a user. On ICP, an issuer is an exposed API that recieves calls from an identity provider and does not trigger any workflows itself. To become an issuer, your canister must implement the issuer API as described in [verifiable credential spec](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md).
+
+Issuers receive credentials requests from relying partys through an identity provider such as Internet Identity (II). The end-user must allow or revoke the communication before the credential is issued.
+
+Internet Identity is used as the identity provider in this tutorial.
+
+## Development of the issuer API
+
+The issuer API consists of four endpoints:
+
+1. [`vc_consent_message`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#1-consent-message): Responds with a message of consent to be displayed to the user to receive their consent on issuing the credentials.
+
+2. [`derivation_origin`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#2-derivation-origin): Returns the URL used to derive the user's principal for the issuer. 
+
+3. [`prepare_credential`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#3-prepare-credential): Check the validity of the request and prepare the actual credential requested.
+
+4. [`get_credential endpoint`](https://github.com/dfinity/internet-identity/blob/main/docs/vc-spec.md#4-get-credential): Issues the actual credential requested by the user.
+
+### Step 1: Obtain a consent message (`vc_consent_message`)
+
+First, the user must send a consent message to the issuer API endpoint `vc_consent_message` through the identity provider. A consent message is an approval from the user to validate that they consent to the issuing of a VC.
+
+An example implementation in Rust can be found below:
+
+```rust
+pub fn get_vc_consent_message_en(
+    credential_spec: &CredentialSpec,
+) -> Result<Icrc21ConsentInfo, Icrc21Error> {
+    match verify_spec_and_get_group_name(credential_spec) {
+        Err(err) => Err(Icrc21Error::ConsentMessageUnavailable(Icrc21ErrorInfo {
+            description: err,
+        })),
+        Ok((group_name, _owner)) => Ok(Icrc21ConsentInfo {
+            consent_message: format!("# \"{}\"", group_name),
+            language: "en".to_string(),
+        }),
+    }
+}
+```
+
+If the response is an approval, the identity provider returns the consent message to the issuer.
+
+### Step 2: Implement a derivation origin (`derivation_origin`)
+
+Next, the issuer API endpoint `derivation_origin` enables support of the [alternative derivation origins](https://internetcomputer.org/docs/current/references/ii-spec#alternative-frontend-origins) feature.
```

Comment:
I would copy what's in the spec:

> `derivation_origin` is called by the identity provider to obtain an URL to be used as the derivation origin for user's principal.

We have seen some confusion regarding this value. Therefore, I would add.

> If the issuer relies on the [Alternative Derivation Origins](https://internetcomputer.org/docs/current/references/ii-spec#alternative-frontend-origins) feature, the endpoint must return the same value for the `derivationOrigin` when logging in.

> If the issuer doesn't rely on Alternative Derivation Origins, then it must return the canister's URL: `https://<issuer-canister-id>.icp0.io`.

Reply:
Revised per feedback